File: src/main/java/com/pharmeasy/consent/dto/EmployeeDto.java
package com.pharmeasy.consent.dto;

import com.pharmeasy.consent.entity.Employee.Role;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.extern.slf4j.Slf4j;

import java.util.List;

import static com.pharmeasy.consent.utils.Constants.regexEmail;

@Slf4j
public record EmployeeDto(
    @NotBlank(message = "Email is mandatory") @Email(message = "Invalid email" +
                                                               " format") @Pattern(
        regexp = regexEmail, message = "Email must be from pharmeasy.in domain"
    ) String email,

    @NotBlank(message = "First name is mandatory") @Pattern(
        regexp = "^[a-zA-Z]+$", message = "First name must contain only letters"
    ) String firstName,

    @Pattern(
        regexp = "^[a-zA-Z]*$",
        message = "Middle name must contain only letters"
    ) String middleName,

    @NotBlank(message = "Last name is mandatory") @Pattern(
        regexp = "^[a-zA-Z]+$", message = "Last name must contain only letters"
    ) String lastName,

    @NotNull(message = "Role is mandatory") Role role,

    List<String> ownedServices,
    List<String> requestedServices,
    List<String> accessibleServices
)
{
}

File: src/main/java/com/pharmeasy/consent/dto/EmployeeDto.kt
package com.pharmeasy.consent.dto

import com.pharmeasy.consent.entity.Employee
import com.pharmeasy.consent.utils.Constants
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import lombok.*
import lombok.extern.slf4j.Slf4j

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Slf4j
class EmployeeDto {
    private var email: @NotBlank(message = "Email is mandatory") @Email(message = "Invalid email format") @Pattern(
        regexp = Constants.regexEmail,
        message = "Email must be from pharmeasy.in domain"
    ) String? = null

    private var firstName: @NotBlank(message = "First name is mandatory") @Pattern(
        regexp = "^[a-zA-Z]+$",
        message = "First name must contain only letters"
    ) String? = null

    private var middleName: @Pattern(
        regexp = "^[a-zA-Z]*$",
        message = "Middle name must contain only letters"
    ) String? = null

    private var lastName: @NotBlank(message = "Last name is mandatory") @Pattern(
        regexp = "^[a-zA-Z]+$",
        message = "Last name must contain only letters"
    ) String? = null

    @Enumerated(EnumType.STRING)
    private var role: @NotNull(message = "Role is mandatory") Employee.Role? = null

    private var ownedServices: List<String>? = null

    private var requestedServices: List<String>? = null
    private var accessibleServices: List<String>? = null
}
